packages/frontend/src/features/catalog/components/CatalogTableListItem.tsx

```diff
@@ -0,0 +1,121 @@
+import { type CatalogField, type CatalogTable } from '@lightdash/common';
+import { ActionIcon, Box, Collapse, Group, Highlight } from '@mantine/core';
+import {
+    IconChevronRight,
+    IconExternalLink,
+    IconTable,
+} from '@tabler/icons-react';
+import React, { useState, type FC } from 'react';
+import { useToggle } from 'react-use';
+import MantineIcon from '../../../components/common/MantineIcon';
+import MantineLinkButton from '../../../components/common/MantineLinkButton';
+
+type Props = {
+    table: CatalogTable & { fields: CatalogField[] };
+    startOpen?: boolean;
+    searchString?: string;
+    isSelected?: boolean;
+    url: string;
+    onClick?: () => void;
+};
+
+export const CatalogTableListItem: FC<React.PropsWithChildren<Props>> = ({
+    table,
+    startOpen = false,
+    searchString = '',
+    isSelected = false,
+    url,
+    onClick,
+    children,
+}) => {
+    const [isOpen, toggleOpen] = useToggle(startOpen);
+    const [hovered, setHovered] = useState<boolean | undefined>(false);
+
+    const handleOpenClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+        event.stopPropagation();
+        toggleOpen();
+    };
+
+    return (
+        <>
+            <Group
+                noWrap
+                position="apart"
+                py="sm"
+                px="xs"
+                sx={(theme) => ({
+                    minHeight: 40,
+                    borderRadius: theme.radius.sm,
+                    padding: theme.spacing.md,
+                    backgroundColor: hovered
```

Comment:
does the background color not change if selected ?

Reply:
It gets a border for now. We could change the style more.

Reply:
Priyanka said she was going to work on styles.